Cython/Utility/MemoryView_C.c

```diff
@@ -347,18 +347,23 @@ static int __Pyx_ValidateAndInit_memviewslice(
     }
 
     /* Check axes */
-    for (i = 0; i < ndim; i++) {
-        spec = axes_specs[i];
-        if (unlikely(!__pyx_check_strides(buf, i, ndim, spec)))
-            goto fail;
-        if (unlikely(!__pyx_check_suboffsets(buf, i, ndim, spec)))
+    if (buf->len > 0) {
+        /* 0-sized arrays do not undergo these checks since their strides are
+         * irrelevant and they are always both C- and F-contiguous
+         */
```

Comment:
C-style multi-line comments are not discarded from the generated C file.
```suggestion
        // 0-sized arrays do not undergo these checks since their strides are
        // irrelevant and they are always both C- and F-contiguous.
```